### ISSUE TITLE
Use blinded flow in validator by default

### ISSUE THREAD

**Issue by dapplion** [ChainSafe/lodestar#5243]
**Is your feature request related to a problem? Please describe.**

- See https://github.com/ChainSafe/lodestar/issues/5226

**Describe the solution you'd like**

For blocks created locally, use blinded block by default to prevent having to send transactions on JSON

**Comment by twoeths** [comment]
I suppose the flow would be, please confirm @dapplion :
- New Rest api to produce local blinded block
- Cache full ExecutionPayload by root
- New Rest api to publish local blinded block, combine with the cached payload above to create the regular SignedBeaconBlock

Two questions I have:
- Need 2 new apis, or new header for existing produce/publish block api at https://github.com/ethereum/consensus-specs , otherwise our validator cannot work with beacon nodes from other clients
- Fallback urls at `vc` may not work if the local blinded block is published to a different beacon node

**Comment by dapplion** [comment]
1. You don't need need APIs, the existing ones used by regular blinded flow should apply. Check with @g11tech . Also feel free to reach out to Teku AFAIK they use this strategy internally.
2. Yes you would need to cache the full payload internally
3. Yes Fallback URLs will not work for an interrupted flow. This is already an existing problem for attestation -> aggregate, so it's okay. When using fallbacks, the validator sends data to all possible nodes, so there's no chance of data loss only loud errors

**Comment by wemeetagain** [comment]
I think this is no longer relevant now that we send blocks in binary format? cc @nflaig

**Comment by nflaig** [comment]
It is still relevant but it's not safe to enable this by default, we have a flag `--blindedLocal` which can be set on the vc but it does not work with other bn and while it does not cause issues there, as it will just return full block, it is dangerous if you have multiple beacon nodes connected as only the bn that produced the block can publish it. Generally, we wanna avoid state in block proposal flow.

As you noted, this issue has been drastically reduced since we use ssz everywhere.

**Comment by nflaig** [comment]
also @philknows I think you mentioned this recently, our count of missed / orphaned blocks is basically 0 now

**Comment by nflaig** [comment]
I would say we should only consider enabling this by default if
- it is standarized on the beacon api to request local block as blinded
- and we use a strategy to publish blinded block to all connected bns

**Comment by philknows** [comment]
> As you noted, this issue has been drastically reduced since we use ssz everywhere.

Correct. The last known issues we had with orphaning/missed blocks on mainnet was resolved by `--blindedLocal` due to how we've setup our infrastructure (separate host BNs from host VCs). We haven't tested removing `--blindedLocal` to see how the SSZ APIs handle now that it's included, but AFAIK, it's really just used by sophisticated setups/users like us who use only Lodestar BNs and VCs. 

Would weakly lean against making it a default with similar considerations mentioned by @nflaig above.

**Comment by nflaig** [comment]
> Would weakly lean against making it a default with similar considerations mentioned by @nflaig above.

Also a lot of setups have this enabled by default in their configuration where we know it's safe to do, for example in eth-docker (combined setup) and dappnode.

If users see high latencies in their block proposal flow we can always suggest to set this flag.
